### PR TITLE
Fixes to Development to Master and refactors to DocumentSelection

### LINKED ISSUE #208: Document Selection Style Updates
Adding MUI styling to the "Document Selection" component (`DocumentSelection.jsx`), which is implemented by multiple other components. This has been kept as a separate issue in order to ease the process and reduce potential bugs.

To begin once all components using Document Selection are complete.

### PR DESCRIPTION
**Pull request by leekahung**
@xscottxbrownx The fixes and refactoring are pretty significant to #234 so I've pulled these to another PR.

Not entirely why the warning messages still appear since they're made in the way that follows the documentation (see https://mui.com/material-ui/react-select/#SelectAutoWidth.js and https://mui.com/material-ui/react-select/#SelectVariants.js)

Let me know if you think it's good.

Other than that, I've pulled the entire FormControl and InputLabel into DocumentSelection, so it's all in one place. It's better to manage, too.

### CODE REVIEW

**Comment by andycwilliams** [comment]
I would like to be alerted about MUI updates a reviewer as well. It is important to reduce conflicting code as much as possible and ensure that so we're all up to date. Thank you.

Also need to alert @masenmatthews as they are working on refactoring style.css into theme.js. They need to be apprised of any changes.

**Comment by leekahung** [comment]
> I would like to be alerted about MUI updates a reviewer as well. It is important to reduce conflicting code as much as possible and ensure that so we're all up to date. Thank you.

I could certainly add you as a reviewer. The main point of this PR was largely related to the comments pointed out in the Dev. to Master PR by Scott, so I've only added him.

As for potential merge conflict, that could be resolved when updating the branches. That should not be too much of an issue?

**Review by xscottxbrownx — APPROVED** [review]
Yeah, all this is by the book of MUI and works - not sure why there's still warnings with the labels.  I've tried a lot of other approaches to try to make the labels work, and it either stays the same or makes it much worse.

---

Have my review/approval of these changes to `<DocumentSelection>`.

---

As a side note, I found this which may help with updating any of these or the clients forms to what we need: https://mui.com/material-ui/react-select/#with-a-dialog

**Comment by leekahung** [comment]
@andycwilliams Yeah, just double checked the merge process locally. There seems to only be one merge conflict if I attempt this (see Merge Changes), but it's not related to DocumentSelection (see second picture).

Screen Shot 2023-06-08 at 14 01 17

Screen Shot 2023-06-08 at 14 05 03

**Comment by andycwilliams** [comment]
It never hurts to have more eyes and communication, even if it's just to keep people up to speed. This is especially so when it'll be part of a large merge with many files changed and it's very easy for things to get lost.

And that branch is not in PR yet so conflicts with that aren't an issue.

It's also good to search for and link to open Issues that PRs might address. In this case, this would close #208.

**Comment by leekahung** [comment]
I'm a little confused to the statement?

> It never hurts to have more eyes and communication, even if it's just to keep people up to speed. This is especially so when it'll be part of a large merge with many files changed and it's very easy for things to get lost.

Which is fine, if the changes seems off after it's incorporated into Development, the PR for #234 is still in progress. So it can still be reviewed over and people should be kept up to speed. If the changes due to this merge produce issues, it can be pointed out during the review process.

File: src/components/Form/CrossPodQueryForm.jsx
Comment (andycwilliams) [inline src/components/Form/CrossPodQueryForm.jsx:1]: I haven't tested this myself yet, but I'm wondering if having nested Box elements might be redundant.

**Comment by leekahung** [comment]
@andycwilliams I guess want I'm curious about, if whether or not if this PR is breaking anything?

File: src/components/Form/CrossPodQueryForm.jsx
Comment (leekahung) [reply src/components/Form/CrossPodQueryForm.jsx:1]: It's used at the moment to keep things aligned within the form.

See with Box.

Screen Shot 2023-06-08 at 14 17 30 (2)

See without Box.

Screen Shot 2023-06-08 at 14 18 25 (2)

Comment (andycwilliams) [reply src/components/Form/CrossPodQueryForm.jsx:1]: Would this not be addressed by fullWidth?

**Review by andycwilliams — APPROVED** [review]
Doesn't seem to conflict. Approving.

File: src/components/Form/CrossPodQueryForm.jsx
Comment (leekahung) [reply src/components/Form/CrossPodQueryForm.jsx:1]: Yeah, I've tried that before. I also thought it'll be redundant to add another Box at the beginning, but we'll have to center the button.

Screen Shot 2023-06-08 at 14 25 08

Think Box is the best compromise for now even though it stretches the button out.

**Pull request merged**